docs/design/access-control.md

```diff
@@ -0,0 +1,93 @@
+# UX for role-based access control
+
+This design document is a proposal for a Dashboard UI that
+allows users to manage Roles, RolesBindings, ClusterRoles, and 
+ClusterRoleBindings in an intuitive way. This document is a 
+formalization of ideas and discussions from #1441.
+
+## Background
+
+Kubernetes now allows cluster admins to use 
+[role-based access control](http://kubernetes.io/docs/admin/authorization/) to secure their clusters. 
+Permissions are managed using the following Kubernetes Resources:
+* Role
+* ClusterRole
+* RoleBinding
+* ClusterRoleBinding
+
+None of these are reflected in the current version of Dashboard UI. There should a simplified, 
+more user-friendly way to manage a cluster's access control by abstracting these raw concepts.
+
+
+## Abstraction / Simplification
+
+The only difference between Roles and ClusterRoles is that Roles apply to a specific namespace 
+and ClusterRoles apply to all namespaces. The same is true for RoleBindings and ClusterRoleBindings. 
+The UI should help users to understand these concepts and user flows should be designed to enforce
+the correct usage. 
+
+NOTE: The design proposed in this document will combine certain concepts:
+* ClusterRoles will be referred to as Roles for the namespace "*"
+* ClusterRoleBindings will be referred to as RoleBindings for the namespace "*"
+
+## Use Cases
+
+* Shortly introduce the concept of roles and bindings to the user
+* Create a Role
+* Bind subjects to a Role
+* Edit a role's rules
+* Unbind a subject from a role
+
+# Design
+
+
+
+## Overview
+
+The overview can be reached by clicking the "Access Control" item in the admin section of the
+left-hand menu. It shows existing Roles/Bindings, some important details for every item and allows
+the user to create a new Role or Binding right away.
+
+![Overview](mockups/21-11-2016-access-control/overview.png)
+
+## Creating a role
+
+Clicking the "+" under the existing Roles will open a dialog. After entering the name and
```

Comment:
This is a tricky part. We've never had such proposal for creating objects. All we had is this +create button, which is mildly useful:
![image](https://cloud.githubusercontent.com/assets/1159999/20600813/bce93aca-b255-11e6-8478-24f2ddcd8b1e.png)

I like the proposal here to move creation to the context of where it is used. We can apply this pattern project wide if it turns out to be good.


Other option would be to add "+create" button to the list toolbar, as this is our long term place to put things related to a list displayed below.  

@normalnorman @danielromlein What do you think re this?